DirectProgramming/DPC++FPGA/Tutorials/Features/experimental/register_map_and_streaming_interfaces/src/register_map_and_streaming_interfaces.cpp

```diff
@@ -0,0 +1,209 @@
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/ext/intel/prototype/interfaces.hpp>
+#include <sycl/sycl.hpp>
+
+#include "exception_handler.hpp"
+
+using ValueT = int;
+// Forward declare the kernel names in the global scope.
+// This FPGA best practice reduces name mangling in the optimization reports.
+class LambdaRegisterMapControlIP;
+class LambdaStreamingControlIP;
+
+// offloaded computation
+ValueT SomethingComplicated(ValueT val) { return (ValueT)(val * (val + 1)); }
+
+/////////////////////////////////////////
+
+struct FunctorStreamingControlIP {
+  // Use the 'conduit' annotation on a kernel argument to specify it to be
+  // a streaming kernel argument.
+  conduit ValueT *input;
+  conduit ValueT *output;
+  // Without the annotations, kernel arguments will be inferred to be streaming
+  // kernel arguments if the kernel control interface is streaming, and
+  // vise-versa.
+  size_t n;
+  FunctorStreamingControlIP(ValueT *in_, ValueT *out_, size_t N_)
+      : input(in_), output(out_), n(N_) {}
+  // Use the 'streaming_interface' annotation on a kernel to specify it to be
+  // a kernel with streaming kernel control signals.
+  streaming_interface void operator()() const {
+    for (int i = 0; i < n; i++) {
+      output[i] = SomethingComplicated(input[i]);
+    }
+  }
+};
+
+struct FunctorRegisterMapControlIP {
+  // Use the 'register_map' annotation on a kernel argument to specify it to be
+  // a register map kernel argument.
+  register_map ValueT *input;
+  // Without the annotations, kernel arguments will be inferred to be register
+  // map kernel arguments if the kernel control interface is register map, and
+  // vise-versa.
+  ValueT *output;
+  // A kernel with register map control can also independently have streaming
+  // kernel arguments, when annotated by 'conduit'.
+  conduit size_t n;
+  FunctorRegisterMapControlIP(ValueT *in_, ValueT *out_, size_t N_)
+      : input(in_), output(out_), n(N_) {}
+  register_map_interface void operator()() const {
+    for (int i = 0; i < n; i++) {
+      output[i] = SomethingComplicated(input[i]);
+    }
+  }
+};
+
+void TestLambdaRegisterMapControlKernel(sycl::queue &q, ValueT *in, ValueT *out,
+                                        size_t count) {
+  // In the Lambda programming model, all kernel arguments will have the same
+  // interface as the kernel control interface.
+  q.single_task<LambdaRegisterMapControlIP>([=] register_map_interface {
+     for (int i = 0; i < count; i++) {
+       out[i] = SomethingComplicated(in[i]);
+     }
+   }).wait();
+
+  std::cout << "\t Done" << std::endl;
+}
+
+void TestLambdaStreamingControlKernel(sycl::queue &q, ValueT *in, ValueT *out,
+                                      size_t count) {
+  // In the Lambda programming model, all kernel arguments will have the same
+  // interface as the kernel control interface.
+  q.single_task<LambdaStreamingControlIP>([=] streaming_interface {
```

Comment:
The sycl compiler has run into a regression that we are tracking. You should try compiling with sycl/trunk/20221102 instead (the last good resource we had).

Reply:
```bash
build $> arc job resources
aclsycltest/2023.1/20221213.2781,tbb/drop/20220124,acds/23.1/36,gcc/7.4.0/2,python/2.7.13,cmake/3.7.2,dev/oneapi_sample_common,sycl/trunk/20221102
```

I still see registers in the lambdastreamingcontrol IP. here is the report in case you want to investigate it yourself:
https://psg-sc-arc.sc.intel.com/p/psg/swip/w/whitepau/oneAPI_Examples/oneAPI-samples.CSR_and_streaming_interfaces/DirectProgramming/DPC++FPGA/Tutorials/Features/experimental/register_map_and_streaming_interfaces/build/register_map_and_streaming_interfaces.fpga_ip_export.prj_2/reports/report.html

This also doesn't just happen when `-fsycl-device-code-split=per_kernel` is used:
https://psg-sc-arc.sc.intel.com/p/psg/swip/w/whitepau/oneAPI_Examples/oneAPI-samples.CSR_and_streaming_interfaces/DirectProgramming/DPC++FPGA/Tutorials/Features/experimental/register_map_and_streaming_interfaces/build/register_map_and_streaming_interfaces.report.prj/reports/report.html

Reply:
Paul, I arc shelled your exact resources and got the right behavior:
https://psg-sc-arc.sc.intel.com/p/psg/swip/w/tiwaria1/streaming_sycl_tut/rtlag-pw.prj/reports/report.html
Did you grab those resources on a new shell?
@intel-jisheng1 can you sync up with Paul offline and check why he is getting the wrong interfaces?

Reply:
weird... now it works.
I withdraw my complaint.